src/pages/workspaces/workspace/analysis/Analyses.test.ts

```diff
@@ -0,0 +1,389 @@
+  })
+
+  it('loads properly with files for an azure workspace', async () => {
+    // Arrange
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const files = [getFileFromPath(`test/${fileName1}` as AbsolutePath), getFileFromPath(`test/${fileName2}` as AbsolutePath)]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      ...defaultUseAnalysisStore,
+      loadedState: { status: 'Ready', state: files }
+    }))
+
+    // Act
+    await act(async () => { // eslint-disable-line require-await
+      render(h(BaseAnalyses, { ...defaultAnalysesProps, workspace: defaultAzureWorkspace }))
+    })
+
+    // Assert
+    screen.getByText('Start')
+    // Table headings
+    screen.getByText('Application')
+    screen.getByText('Name')
+    screen.getByText('Last Modified')
+    screen.getByPlaceholderText('Search analyses')
+    screen.getByText(fileName1)
+    screen.getByText(fileName2)
+  })
+
+  it('Sorts analysis table properly', async () => {
+    // Arrange
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const file1Date = new Date().getTime()
+    const file2Date = file1Date + 10000
+    const files = [{ ...getFileFromPath(`test/${fileName1}` as AbsolutePath), lastModified: file1Date }, { ...getFileFromPath(`test/${fileName2}` as AbsolutePath), lastModified: file2Date }]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      ...defaultUseAnalysisStore,
+      loadedState: { status: 'Ready', state: files }
+    }))
+
+    // Act
+    await act(async () => { // eslint-disable-line require-await
+      render(h(BaseAnalyses, defaultAnalysesProps))
+    })
+
+    const analysisRowsBeforeSort: HTMLElement[] = await screen.findAllByRole('row')
+
+    await act(async () => {
+      const nameTableHeader = screen.getByText('Name')
+      await userEvent.click(nameTableHeader)
+    })
+
+    // Assert visual state before sorting
+    expect(analysisRowsBeforeSort.length).toBe(3)
+    expect(analysisRowsBeforeSort[1].textContent).toContain(fileName2)
+    expect(analysisRowsBeforeSort[2].textContent).toContain(fileName1)
+
+    // Assert files have inverted in table after sort
+    const analysisRowsAfterSort: HTMLElement[] = await screen.findAllByRole('row')
+    expect(analysisRowsAfterSort.length).toBe(3) // 2 files plus 1 header row = 3
+    expect(analysisRowsAfterSort[1].textContent).toContain(fileName1)
+    expect(analysisRowsAfterSort[2].textContent).toContain(fileName2)
+  })
+
+  it('Properly navigates to launcher on clicking an analysis', async () => {
+    // Arrange
+    const navGoToPathObservable = jest.fn()
+    asMockedFn(goToPath).mockImplementation(navGoToPathObservable)
+
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const files = [getFileFromPath(`test/${fileName1}` as AbsolutePath), getFileFromPath(`test/${fileName2}` as AbsolutePath)]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
```

Comment:
Nit: For mocks that return a constant, `mockReturnValue(value)` is a bit more concise than `mockImplementation(() => value)`.